src/dishka/integrations/base.py

```diff
@@ -73,6 +115,8 @@ async def auto_injected_generator(
         )
         container = container_getter(args, kwargs)
         async with container(additional_context) as container:
+            if isinstance(dependencies, ParameterDependencyResolver):
```

Comment:
let's have 2 spearate objects. One is used to bind, another - to iterate over deps. So we can simplify check with `if binder is not None`